source/StartConditions.cpp

```diff
@@ -148,13 +144,49 @@ void StartConditions::Load(const DataNode &node)
 			else
 				child.PrintTrace("Skipping unrecognized attribute:");
 		}
+		else if(key == "on")
+		{
+			if(child.Token(1) == "display")
+				LoadState(child, StartState::VISIBLE);
+			else if(child.Token(1) == "reveal")
+				LoadState(child, StartState::REVEALED);
+			else if(child.Token(1) == "unlock")
+				LoadState(child, StartState::UNLOCKED);
```

Comment:
This would remove the need to set the core data after loading is finished.
```suggestion
			else if(child.Token(1) == "unlock")
			{
				// The "on unlock" info is the "true" info of these StartConditions, and so
				// some info may need to be loaded into the CoreStartData as well.
				CoreStartData::LoadChild(child, add, remove)
				LoadState(child, StartState::UNLOCKED);
			}
```

Reply:
done